### PR TITLE
Add log message when local input's filter does not match any files

### PR DESCRIPTION
**Pull request by loquisgon**
### Expected behavior

When the filter for local inputSource does not match any file a log message should be output at `INFO` level indicating that "no files matched the filter expression:.

### Actual behavior

When the filter for local inputSource does not match any file there is no log message, the task report simply indicates that no rows were parsed.

### Key changes:
Added a logger & log message to class `LocalInputSource.java` to match the expected behavior as outlined above.

### CODE REVIEW

File: core/src/main/java/org/apache/druid/data/input/impl/LocalInputSource.java [hunk 1]
```diff
@@ -151,6 +154,16 @@ public int estimateNumSplits(InputFormat inputFormat, @Nullable SplitHintSpec sp
...
+          baseDir.getAbsoluteFile(),
+          fileFilter,
+          TrueFileFilter.INSTANCE
+      );
+      if (!fileIterator.hasNext()) {
+        // base dir & filter are guaranteed to be non-null here
+        // (by construction and non-null check of baseDir a few lines above):
+        log.info("Local inputSource filter [%s] for base dir [%s] did not match any files", filter, baseDir);
+      }
       return FileUtils.iterateFiles(
```
Comment (loquisgon) [inline core/src/main/java/org/apache/druid/data/input/impl/LocalInputSource.java:167]: Forgot to re-use fileIterator here, I will fix this.

Comment (loquisgon) [inline core/src/main/java/org/apache/druid/data/input/impl/LocalInputSource.java:167]: Forgot to re-use fileIterator, I will fix it right away

File: core/src/main/java/org/apache/druid/data/input/impl/LocalInputSource.java [hunk 2]
```diff
@@ -151,6 +154,16 @@ public int estimateNumSplits(InputFormat inputFormat, @Nullable SplitHintSpec sp
...
+      Iterator<File> fileIterator = FileUtils.iterateFiles(
+          baseDir.getAbsoluteFile(),
+          fileFilter,
+          TrueFileFilter.INSTANCE
+      );
+      if (!fileIterator.hasNext()) {
+        // base dir & filter are guaranteed to be non-null here
+        // (by construction and non-null check of baseDir a few lines above):
+        log.info("Local inputSource filter [%s] for base dir [%s] did not match any files", filter, baseDir);
+      }
```
Comment (loquisgon) [inline core/src/main/java/org/apache/druid/data/input/impl/LocalInputSource.java:166]: Forgot to re-use fileIterator below, I will fix it right away

Comment (loquisgon) [reply core/src/main/java/org/apache/druid/data/input/impl/LocalInputSource.java:166]: Fixed

**Review by suneet-s — APPROVED** [review]
LGTM - travis failures appear to be because of an issue with something in master

**Review by a2l007 — APPROVED** [review]
This is useful! Thanks @loquisgon 
I've restarted the build to see if it helps.

**Comment by suneet-s** [comment]
The leadership test failures were fixed in #10846 so I think it's ok to overrule Travis for this change.

**Pull request merged**